docs/source/_template/class.rst

```diff
@@ -0,0 +1,9 @@
+.. role:: hidden
+    :class: hidden-section
+.. currentmodule:: {{ module }}
```

Comment:
@NivekT : Seeing the following warning:
```
/Users/wxie/Code/torcharrow/docs/source/_template/class.rst:3: ERROR: Error in "currentmodule" directive:
maximum 1 argument(s) allowed, 3 supplied.

.. currentmodule:: {{ module }}
WARNING: invalid signature for autoclass ('{{ name }}')
WARNING: don't know which module to import for autodocumenting '{{ name }}' (try placing a "module" or "currentmodule" directive in the document, or giving an explicit module name)
```

Wondering if there is anything I am not configured correct? :(

Reply:
That is a template file, and you can use it with `autosummary`. The file where you call needs to have "`.. currentmodule:: ` specified. It doesn't seen like you are using `class.rst` right now?

```rst
.. currentmodule:: torchdata.datapipes.iter

......

.. autosummary::
    :nosignatures:
    :toctree: generated/
    :template: datapipe.rst

    Bz2FileLoader
    Decompressor
    RarArchiveLoader
    TarArchiveLoader
```

Reply:
@NivekT : Right. I see for now I am only using `:template: function.rst`. But see the similar warning. Basically it works fine when used in `autosummary`. But somehow Sphinix try to also parse it alone so such warnings emitted?

Reply:
Yep, I would probably exclude it if you are not using it.